retrieve_files.sh

```diff
@@ -0,0 +1,171 @@
+#!/bin/bash
+#set -x
+
+###################################################################################################################
+#
+#Script to download,extract, and parse PTAB pdf files from USPTO bulk data site. 
+#
+#argument options:
+#           (1) pass -d OR --date AND YYYYMMDD to download files from a specific date on
+#           (2) pass -a OR --all to download all files
+#           (3) pass -n OR --none to skip downloading of files and execute unzipping process
+#
+###################################################################################################################
+
+#create lock file
+function lock ( ) {
+  if "$1" == true
+  then
+    if [ -f ${lockFile} ]
+    then
+      echo
+      echo "${lockFile} exits."
+      echo "Looks like a copy of $scriptName is already running... "
+      echo "Abort"
+      log "ERR" " Already running, please check logs for reason... "
+      exit 0
+    else
+      touch ${lockFile}
+    fi
+  else
+    if [ -f $lockFile ] ; then
+      rm $lockFile
+    fi
+  fi
+}
+
+function usage
+{
+    echo "usage: ./retrieve_files.sh [[-d YYYMMDD]| [-a] | [-n]]"
+}
+
+function log()
+{
+  type=$1
+  message=$2
+
+  if [ $type = "WARN" -o $type = "ERR" ]
+  then
+    echo >> $statusDirectory/err-$processingTime
+    echo -e "$type: `date +%Y.%m.%d-%H:%M:%S` -- $scriptName -- $message" >> $statusDirectory/err-$processingTime
+  fi
+  #
+  # always write into log file
+  #
+  echo
+  echo -e "$type: `date +%Y.%m.%d-%H:%M:%S` -- $scriptName -- $message"
+  echo >> $statusDirectory/log-$processingTime
+  echo -e "$type: `date +%Y.%m.%d-%H:%M:%S` -- $scriptName -- $message" >> $statusDirectory/log-$processingTime
+}
+
+#=============================== MAIN BODY OF SCRIPT ===============================
+
+##### Constants
+processingTime=`date +%Y%m%d-%H%M%S`
+scriptName=$0
+statusDirectory=logs
+baseURL="https://bulkdata.uspto.gov/data2/patent/trial/appeal/board/"
+dropLocation="files/`date +%Y%m%d`"
+declare -i fileDate
+retrieveAll=false
+retrieveNone=false
+lockFile="/tmp/file_download.lck"
+
+touch $statusDirectory/log-$processingTime
+
+lock true
+
+case "$1" in
+-d | --date )
+  shift
+  echo "$1"
+  if date -d $1 >/dev/null 2>&1
+  then
+    fileDate=$1
+  else
+    log "ERR" "date passed in is not valid: $1"
+    lock false
+    exit 1
+  fi
+  ;;
+-a | --all )
+  retrieveAll=true
+  ;;
+-n | --none )
+  retrieveNone=true
+  ;;
+-h | --help )
+  usage
+  lock false
+  exit
+  ;;
+* )
+  usage
+  lock false
+  log "ERR" "argument passed in is not valid: $1"
+  exit 1
+esac
+     
+#create directory for downloaded files(if does not exist already)
+mkdir -p $dropLocation
+
+currentdate=$(date +%Y%m%d) 
+
+DATE="`date`"
```

Comment:
Looks like this $DATE and the one near the end are not used except in the line that follows. How about just using command substitution in the log line, ala:

```bash
log "INFO" "-[JOB START] $(date): ------------"
```